src/js/firstparties/facebook.js

```diff
@@ -0,0 +1,63 @@
+// Adapted from https://github.com/mgziminsky/FacebookTrackingRemoval
+(function() {
+let fb_wrapped_link = "a[href*='facebook.com/l.php?'";
+
+function findInAllFrames(query) {
+  let out = [];
+  document.querySelectorAll(query).forEach((node) => {
+    out.push(node);
+  });
+  Array.from(document.getElementsByTagName('iframe')).forEach((iframe) => {
+    try {
+      iframe.contentDocument.querySelectorAll(query).forEach((node) => {
+        out.push(node);
+      });
+    } catch (e) {
+      // pass on cross origin iframe errors
+    }
+  });
+  return out;
+}
+
+// remove all attributes from a link except for class and ARIA attributes
+function cleanAttrs(elem) {
+  for (let i = elem.attributes.length - 1; i >= 0; --i) {
+    const attr = elem.attributes[i];
+    if (attr.name !== 'class' && !attr.name.startsWith('aria-')) {
+      elem.removeAttribute(attr.name);
+    }
+  }
+}
+
+// Remove excessive attributes and event listeners from link a and replace
+// its destination with href
+function cleanLink(a) {
+  let href = new URL(a.href).searchParams.get('u');
+  cleanAttrs(a);
+  a.href = href;
+  a.target = "_blank";
+  a.addEventListener("click", function (e) { e.stopPropagation(); }, true);
+  a.addEventListener("mousedown", function (e) { e.stopPropagation(); }, true);
+  a.addEventListener("mouseup", function (e) { e.stopPropagation(); }, true);
+  a.addEventListener("mouseover", function (e) { e.stopPropagation(); }, true);
+  a.addEventListener("mousewheel", function (e) { e.stopPropagation(); }, true);
+}
+
+// unwrap wrapped links in the original page
+findInAllFrames(fb_wrapped_link).forEach((link) => {
+  cleanLink(link);
+});
+
+// Execute redirect unwrapping each time new content is added to the page
+new MutationObserver(function(mutations) {
```

Comment:
If we do introduce any performance problems, it will probably be because of this observer. We could instead apply the same approach we take on Twitter: rerun `findInAllFrames` every two seconds. Imperfect but probably more predictable.

Reply:
This is just a thought if issues do come up, no need to change now unless we discover issues ourselves.

Reply:
I wrote this to check every 2 seconds at first, but found that it was almost useless about half the time because of the hover-to-replace-link feature facebook has. If you scroll down to the bottom of a page, new content loads, and your mouse happens to hover over the new content before 2 seconds elapse, then the hover event will trigger and facebook will replace the l.php URL with the original one, making our current selector useless.

I think the mutation observer is by far the easiest way to prevent that, and haven't noticed any problems yet.